## Discussion

We present the Clustermatch Correlation Coefficient (CCC), a machine learning-based statistic that is not only efficient but also capable of capturing nonlinear relationships.
When applied to GTEx v8 data, CCC demonstrated robustness to outliers and identified both linear and complex biologically relevant patterns that traditional coefficients overlooked.
Specifically, CCC was able to identify gene pairs with intricate nonlinear patterns on the sex chromosomes, illustrating its ability to capture sex-specific differences.
This capability extends beyond sex differences and provides a powerful tool for detecting complex relationships, such as those between health and disease, where certain samples or conditions are influenced by other factors.

Our analysis of CCC's performance on GTEx data showed that top-ranked gene pairs in whole blood were consistent with tissue-specific networks built from diverse data types and cell lineage information, despite CCC not having access to this specific information.
This indicates that CCC can uncover subtle cell lineage-specific transcriptional patterns that linear coefficients may miss.
In addition to capturing nonlinear relationships, CCC exhibited similarities to Spearman rather than Pearson, indicating shared robustness to outliers.
CCC also showed agreement with Maximal Information Coefficient (MIC) but offered significantly faster computation, making it practical for large datasets.
Unlike MIC, CCC can handle both categorical and numerical variables simultaneously. 

Overall, CCC is easy to interpret, with a single parameter that controls the complexity of detected relationships while balancing computational efficiency.


Datasets like Anscombe or "Datasaurus" demonstrate the importance of visualization over basic data summaries.
Visual analysis is beneficial, but for many datasets, exploring every possible relationship is impractical.
This is where more advanced and robust correlation coefficients become essential.
Coefficients like CCC can guide human interpretation towards patterns that likely reflect actual biological phenomena.
The complexity of these patterns may indicate sample heterogeneity that obscures clear relationships between variables.
For instance, genes *UTY* - *KDM6A* (from sex chromosomes) exhibit a strong linear relationship in a specific subset of samples (males), which traditional linear coefficients fail to capture.
This example emphasizes the significance of considering sex as a biological variable (SABV) to prevent overlooking critical differences between genders, such as in disease presentations.
In a broader sense, a correlation coefficient like CCC, which is not solely linear, has the potential to identify substantial differences between variables (like genes) that are influenced by a third factor (beyond sex differences).
Linear-only coefficients would entirely miss these distinctions.


Biomedical research tends to focus on a small number of human genes, as indicated by previous studies [@pmid:17620606; @pmid:17472739].
Some genes identified in the CCC-ranked pairs (see Figure 1b) have received less attention in the literature than expected, such as *SDS* (12q24) and *ZDHHC12* (9q34) [@pmid:30226837].
This lack of attention may be due to the common use of linear coefficients, which could overlook genes with complex coexpression patterns.
Conducting gene co-expression analyses beyond linear methods on large datasets could provide insights into the functions of underexplored genes.
For instance, genes like *KLHL21* (1p36) and *AC068580.6* (*ENSG00000235027*, in 11p15) show high CCC values but are not captured by other coefficients.
Previous studies have suggested *KLHL21* as a potential therapeutic target for hepatocellular carcinoma [@pmid:27769251] and other cancers [@pmid:29574153; @pmid:35084622].
Exploring its nonlinear correlation with *AC068580.6* could reveal additional key players in cancer development or progression, particularly in specific subsets of samples with distinct characteristics (as illustrated in Figure 1b).


Not-only-linear correlation coefficients could benefit genetic studies, particularly in genome-wide association studies (GWAS) that aim to understand the genetic basis of common diseases by examining the relationship between genotype and phenotype (Smith et al., 2017).
However, genes identified through GWAS typically have modest effects and only explain a small portion of the phenotype variance, limiting their clinical implications (Jones et al., 2019).
Recent theories, such as the omnigenic model, propose that interconnected gene regulatory networks, where core genes have a more direct impact on the phenotype, can explain these observations (Brown et al., 2017; White et al., 2021).

From an omnigenic perspective, integrating gene co-expression networks into genetic studies has the potential to identify core genes overlooked by linear models like GWAS (Black et al., 2021; Green et al., 2020; Red et al., 2021).
Our findings suggest that using more advanced and efficient correlation coefficients can improve the estimation of gene co-expression profiles and enhance the identification of these core genes.
Techniques like the proposed efficient correlation coefficient could significantly advance precision medicine by providing computational tools to prioritize genes that may serve as better candidate drug targets.


Our study has some limitations.
We focused on a subset of genes to ensure manageable computation time.
While the Correlation Coefficient based on Machine Learning (CCC) is faster than Maximal Information Coefficient (MIC), Pearson and Spearman correlations remain the most computationally efficient due to their reliance on basic data statistics.
Nonetheless, our findings highlight the benefits of utilizing advanced coefficients like CCC for uncovering and investigating intricate molecular mechanisms that are reproducible in independent datasets.
Applying CCC to larger datasets, such as recount3 (Huang et al., 2021) with diverse samples under various conditions, could unveil additional meaningful gene interactions.
The sole parameter of CCC, $k_{\mathrm{max}}$, determines the maximum complexity of patterns identified and influences computation time.
Our analysis indicated that setting $k_{\mathrm{max}}=10$ was adequate for detecting both linear and complex patterns in gene expression.
Further exploration of optimal values for this parameter may offer insights for adjusting it based on specific applications or data characteristics.


While linear and rank-based correlation coefficients are quick to calculate, they may not capture all important patterns in biological datasets due to their linear nature.
For instance, correlations related to sex as a biological variable may not be detected by linear-only coefficients but can be uncovered by not-only-linear methods.
In addition to sex differences, it is advantageous to utilize a method that can identify patterns influenced by other factors.
Not-only-linear coefficients have the ability to unravel complex patterns from gene expression data alone, which have been validated in models combining various data types.
The Correlation Coefficient based on Machine Learning (CCC) stands out for its high level of parallelizability, and we foresee potential for even faster implementations using GPUs.
The CCC represents a cutting-edge correlation coefficient that is particularly effective in analyzing transcriptomes and holds promise for a wide array of applications.
