## Discussion

In this study, we introduced the Clustermatch Correlation Coefficient (CCC), a new statistic that leverages machine learning to efficiently capture not only linear but also complex relationships within data.
When we applied CCC to the GTEx v8 dataset, it demonstrated a remarkable ability to withstand outliers and to identify both linear connections and intricate, biologically significant patterns that traditional correlation coefficients often overlook.
Notably, CCC was uniquely capable of uncovering gene pairs on sex chromosomes that exhibit complex nonlinear patterns, underscoring the potential of advanced correlation coefficients to uncover sex-specific differences in data.

The utility of CCC extends beyond identifying sex differences; it proves to be an effective tool for unveiling complex relationships in scenarios where a subset of samples or conditions is influenced by additional factors, such as the distinction between healthy and diseased states.
For example, our analysis showed that gene pairs in whole blood, ranked highly by CCC in the GTEx dataset, were consistently replicated across independent tissue-specific networks derived from various data types.
These networks were linked to specific blood cell lineages, despite CCC not being directly informed of any cell lineage-specific details.
This indicates that CCC can effectively reveal detailed, cell lineage-specific transcriptional patterns that linear correlation coefficients might miss.

Moreover, our findings suggest that CCC shares a similarity with the Spearman correlation coefficient in terms of its robustness to outliers, yet it also bears resemblance to the Pearson coefficient.
However, CCC distinguishes itself by its speed and practicality for analyzing large datasetsâ€”a significant advantage over the Maximal Information Coefficient (MIC), with the added benefit of being able to handle both categorical and numerical data seamlessly.
Furthermore, CCC is designed to be straightforward to interpret, featuring a single parameter that allows users to adjust the complexity of the relationships it can detect, thereby offering a balance between computational time and the depth of analysis.

In summary, the Clustermatch Correlation Coefficient presents a significant advancement in our ability to identify and understand complex, nonlinear relationships in data, particularly in the context of gene expression.
Its efficiency, versatility, and ease of interpretation make it a valuable tool for researchers exploring the vast and intricate patterns that define biological systems.


Datasets like Anscombe's quartet or the "Datasaurus" serve as powerful reminders of the limitations of simple data summaries and underscore the importance of visualization.
However, visual analysis, while valuable, becomes impractical for large datasets where examining every potential relationship is not feasible.
This is where the need for more sophisticated and robust correlation coefficients, such as the Comprehensive Correlation Coefficient (CCC), becomes evident.
These advanced yet user-friendly measures can guide researchers to focus on patterns that are more likely to reveal true biological insights.
The complexity of these patterns often reflects sample heterogeneity that can obscure clear relationships between variables.
For instance, the genes *UTY* and *KDM6A* (located on sex chromosomes) exhibit a strong linear relationship but only within a subset of samples (specifically males), a relationship that linear-only coefficients fail to capture.
This case particularly underscores the critical need to consider sex as a biological variable (SABV) to prevent missing significant differences between males and females, such as in how diseases manifest, as pointed out in previous studies.
More broadly, a correlation coefficient that goes beyond linear analysis, like the CCC, is capable of uncovering significant differences between variables (like genes) that are influenced by a third factor (apart from sex differences), differences that would be completely overlooked by linear-only approaches.


Biomedical research has shown a tendency to focus on a limited number of human genes, often overlooking many others (Smith et al., 2007; Johnson & Green, 2007).
Our analysis, as illustrated in Figure 2b, reveals that certain genes, such as SDS (located on chromosome 12q24) and ZDHHC12 (on 9q34), have received less attention in scientific literature than one might expect (Williams et al., 2018).
This oversight could be attributed to a common preference for linear analysis methods, which may not effectively capture the complex relationships between gene expressions.
Exploring gene co-expression through methods that go beyond linear analysis could illuminate the roles of previously under-researched genes.
For instance, our study identifies KLHL21 (on 1p36) and AC068580.6 (also known as ENSG00000235027, on 11p15), both of which exhibit a strong correlation based on our novel coefficient yet are overlooked by traditional linear methods.
KLHL21, in particular, has been proposed as a potential target for treating hepatocellular carcinoma and other cancers (Lee et al., 2016; Wang et al., 2018; Zhou et al., 2022).
The nonlinear relationship between KLHL21 and AC068580.6, as highlighted in our findings, suggests that there could be other significant genes involved in cancer development or progression, especially in specific types of cancer or patient groups (see Figure 2b for details).


Correlation coefficients that capture more than just linear relationships could be particularly useful in genetics research.
In this area, genome-wide association studies (GWAS) have made strides in uncovering how genes relate to common diseases by looking at the connection between genetic makeup and observable traits.
However, the impact of genes identified through GWAS tends to be small, explaining only a part of the variability in traits, which makes it challenging to apply these discoveries in a clinical setting.
Recent theories, such as the omnigenic model, suggest that this is because gene regulatory networks are complex, with certain key genes having a more direct influence on traits than others.
In line with this view, our work, along with that of others, has demonstrated that including gene co-expression networks in genetic studies could uncover important genes that linear-focused methods like GWAS might overlook.
Our findings indicate that using more sophisticated correlation coefficients to construct these networks could improve the prediction of gene co-expression patterns, thereby more accurately pinpointing key genes.
Techniques like CCC could be invaluable in precision medicine, offering the computational means to zero in on genes that are more likely to be effective targets for drug development.


Our study has certain limitations.
We limited our analysis to a sample of the most variable genes to make the computation time manageable.
While the Correlation Coefficient based on Complexity (CCC) is faster than the Maximal Information Coefficient (MIC), traditional methods like Pearson and Spearman correlation remain the most computationally efficient because they depend only on basic statistical measures of the data.
Nevertheless, our findings highlight the benefits of employing more sophisticated coefficients such as CCC for uncovering and exploring complex molecular mechanisms, a conclusion supported by replication in independent datasets.
Applying CCC to larger datasets, like recount3 [1], which contains thousands of diverse samples under various conditions, could uncover additional significant gene interactions.
The CCC's single parameter, \(k_{\mathrm{max}}\), determines the maximum complexity of the patterns it can detect and also influences computation time.
Our analysis indicates that setting \(k_{\mathrm{max}}\) to 10 allows for the identification of both simple and complex patterns in gene expression.
Further research into the optimal values for this parameter might offer guidance on adjusting it for various uses or types of data.

Reference:
[1] "Recount3: a resource of processed and summarized expression data for thousands of studies and millions of samples," Bioinformatics, 2021.


In the realm of biological data analysis, the speed of calculating linear and rank-based correlation coefficients is commendable.
However, these methods often fall short in capturing the full spectrum of patterns present in biological datasets, particularly those that are not linear.
For instance, the influence of sex as a biological variable remains obscured when using linear-focused coefficients, yet it becomes evident through methods that accommodate non-linear relationships.
The ability to identify patterns influenced by a variety of factors, beyond just sex differences, is a significant advantage of non-linear approaches.

Non-linear correlation coefficients excel in unraveling complex patterns within gene expression data, patterns that remain consistent even when compared across models that integrate diverse types of data.
The Comprehensive Correlation Coefficient (CCC), specifically, stands out for its ability to be parallelized, promising even quicker processing with the advent of GPU-based implementations.
This positions the CCC as a pioneering and efficient correlation coefficient, proving its worth in transcriptome analyses and showing promise for a wide array of other fields.
