### Gene expression data and preprocessing {#sec:data_gtex}

We obtained the GTEx version 8 (v8) dataset, which encompasses various tissues, and performed normalization using the Transcripts Per Million (TPM) method.
Our primary focus was on the whole blood dataset due to its substantial sample size, comprising 755 individual samples.
To mitigate the potential bias towards genes with high expression levels, we standardized the expression data of these samples using the logarithmic transformation $log(x + 1)$.
Following this transformation, we identified and selected the top 5,000 genes exhibiting the highest variance within the whole blood dataset.

For these 5,000 genes, we conducted a comprehensive analysis by calculating four distinct correlation coefficients: Pearson, Spearman, Mutual Information-based Criterion (MIC), and Concordance Correlation Coefficient (CCC).
These calculations were performed across all 755 samples using the TPM-normalized data, resulting in the generation of a pairwise similarity matrix with dimensions of 5,000 x 5,000.
This matrix encapsulates the pairwise relationships between genes based on their expression patterns across the sampled population.

The mathematical expressions for the correlation coefficients are as follows:

For Pearson correlation coefficient:

$$
\rho_{X,Y} = \frac{cov(X, Y)}{\sigma_X \sigma_Y}
$$
{#eq:Pear
