### Maximal Information Coefficient (MIC) {#sec:methods:mic}

To estimate the Maximal Information Coefficient (MIC), we employed the Python package `minepy` (version 1.2.5), which is available at [@doi:10.1093/bioinformatics/bts707; @url:https://github.com/minepy/minepy].
This package facilitates the computation of the MIC coefficient, a measure designed to capture both linear and non-linear associations between variables.
In our study, specifically within the GTEx v8 dataset for whole blood samples, we utilized MIC<sub>e</sub>, an enhanced version of the original MIC as introduced by Reshef et al.
[@Reshef2016].
The default settings for this analysis were `alpha=0.6`, `c=15`, and `estimator='mic_e'`, in line with the recommended parameters for optimal performance.

To efficiently handle the computational demands of calculating the MIC across numerous gene expression pairs, we leveraged the `pairwise_distances` function from the `scikit-learn` library [@Sklearn2011].
This approach allowed us to parallelize the computation, significantly reducing the time required to process the extensive GTEx dataset.
Furthermore, to thoroughly assess the computational complexity of our methods, we conducted analyses using both the original MIC, setting the parameter to `estimator='mic_approx'`, and the improved MIC<sub>e</sub>, with `estimator='mic_e'` specified.
These analyses, detailed in the Supplementary Material ([Supplementary Material](#sec:time_test)), provide insights into the computational efficiency and scalability of our approach when applied to large-scale genomic datasets.
